GVRf/Framework/framework/src/main/java/org/gearvrf/x3d/x3dTandLShaderTest.java

```diff
@@ -0,0 +1,254 @@
+            + "        endColor = pixelColor * lightContribution * intensity * lightColor;\n"
+            + "        endColor += specularlightCalculation(pixelNormal, vectorLightToPixel) * u_specularcolor;\n"
+            + "  	}\n"
+            + "  	else if ( (angleLightDirectionToPixel >= cutOffAngleCos) && (beamWidthCos > cutOffAngleCos) ) {\n"
+            + "  	   lightContribution = angleLightToNormal * (angleLightDirectionToPixel - cutOffAngleCos) / (beamWidthCos - cutOffAngleCos) * attenuationValue;\n"
+            + "        endColor = pixelColor * lightContribution * intensity * lightColor;\n"
+            + "        endColor += specularlightCalculation(pixelNormal, vectorLightToPixel) * u_specularcolor;\n"
+            + "  	}\n"
+            + "    endColor = min(endColor, 1.0);\n"
+            + "  }\n"
+            + "  return endColor;\n"         
+            + "}\n";         
+
+    private final String FRAGMENT_SHADER_DmainBgn = "" //
+            + "void main() {\n" //
+            + "  vec3  lightdir = normalize(v_position);\n"
+            + "  vec3  pixelnormal = normalize(v_normal);\n"
+            + "  vec3 pixelcolor = texture2D(texture, texturecoord).rgb * u_diffusecolor;\n"
+            + "  vec3 lightContribution = vec3(0.0, 0.0, 0.0);\n"
+            + "  bool lightsPresent = false;\n"
+    		+ "  vec3 fragmentcolor = vec3(0.0, 0.0, 0.0);\n";
+
+    private final String FRAGMENT_SHADER_DmainEnd = "" //
+            + "  if (!lightsPresent) fragmentcolor = texture2D(texture, texturecoord).rgb;\n" //
+            + "  else fragmentcolor = min(fragmentcolor + u_emissivecolor, 1.0);\n" //
+            + "  gl_FragColor = vec4( fragmentcolor, 1.0 );\n" //
+            + "}\n";
+
+    public String FRAGMENT_SHADER_Lights = ""; //
+
+
+
+    private GVRCustomMaterialShaderId mShaderId;
+    private GVRMaterialMap mCustomShader = null;
+    private GVRMaterialShaderManager shaderManager = null;
+
+    public x3dTandLShaderTest(GVRContext gvrContext) {
+        shaderManager = gvrContext
+                .getMaterialShaderManager();
+    }
+
+
+    public void appendFragmentShaderLights(String lightString) {
+    	FRAGMENT_SHADER_Lights += lightString;
+    }
+
+    public void setCustomShader() {
+ 
+        String FRAGMENT_SHADER_Lights_Present = "lightsPresent = true;\n"; //
+        
+    	if (FRAGMENT_SHADER_Lights.length() == 0) {
+    		FRAGMENT_SHADER_Lights_Present = "lightsPresent = false;\n";
+    	}
+
+        String FRAGMENT_SHADER = "" //
+        		+ FRAGMENT_SHADER_Avars
+        		+ FRAGMENT_SHADER_BspecularlightContribution
+        		+ FRAGMENT_SHADER_Cdirectionallight
+        		+ FRAGMENT_SHADER_Cpointlight
+        		+ FRAGMENT_SHADER_Cspotlight
+        		+ FRAGMENT_SHADER_DmainBgn
+        		+ FRAGMENT_SHADER_Lights_Present
+        		+ FRAGMENT_SHADER_Lights
+                + FRAGMENT_SHADER_DmainEnd;
+        
+        /*
+        Log.e("Mitch -- Vtx", VERTEX_SHADER);
```

Comment:
x3dTandLShaderTest is a pure X3D shader, based on formulas for lighting, materials and texture.  This is currently used to calibrate shader settings between it and the Universal Shader shared by Jasssimp and LoadModel functions, and for performance testing.  Eventually, this code will likely be removed or used to support a pure x3d shader.